### ISSUE TITLE
Codegen nested brackets comment bug

### ISSUE THREAD

**Issue by boyan-soubachov** [stretchr/testify#860]
`Codegen` has a bug when the docstring contains nested brackets (parenthesis). This needs to be fixed as it manifests into inconsistent & incorrect docstrings for our auto-generated functions.
Example: `InDelta`

**Comment by glesica** [comment]
We probably want to keep this around since it's a longer-term problem, so I reopened it.

**Comment by boyan-soubachov** [comment]
Fair point. Thought I’d keep it simple for now but the fix is a bit more involved.

**Comment by glesica** [comment]
Yeah, good call.
